### PR TITLE
Add perspective option.

### PR DESCRIPTION
**Pull request by lukemacausland**

### CODE REVIEW

**Review by akingtorq — APPROVED** [review]
Code changes look good... in the README "specific" should be "specify" I think.

File: README.md [hunk 1]
```diff
@@ -110,3 +110,13 @@ system_roles:
 ```
 
 > **Note:** The default behavior for pimcore is that if no allowed class/doc types are specified, then _all_ classes and doc types are allowed. If you need to restrict all creation, you may need to configure it at the workspace level.
+
+### Allowed Perspectives
+
+You can specific the allowed perspectives on a role by adding perspectives to an array of strings:
```
Comment (lukemacausland) [inline README.md]:
```suggestion
You can specify the allowed perspectives on a role by adding perspectives to an array of strings:
```

**Comment by lukemacausland** [comment]
> Code changes look good... in the README "specific" should be "specify" I think.

Updated

File: src/Command/RoleCreatorCommand.php [hunk 2]
```diff
@@ -209,4 +210,14 @@ private function findDocWithName(string $name, array $docTypes): ?DocType
...
     }
+
+    private function applyPerspectives(Role $role, array $roleProperties): void
+    {
+        if (!key_exists("perspectives", $roleProperties)) {
+            $role->setPerspectives([]);
+            return;
+        }
+
+        $role->setPerspectives($roleProperties["perspectives"]);
```
Comment (rjjackson22) [inline src/Command/RoleCreatorCommand.php]: I believe the null coalescing operator works with array accessing now, so you could simplify this to:
```suggestion
        $role->setPerspectives($roleProperties["perspectives"] ?? []);
```
and if "perspectives" isn't in `$roleProperties` it won't actually throw an exception. Also this only works so long as you trust that `$roleProperties["perspectives"]` is non null

**Review by rjjackson22 — APPROVED** [review]

File: src/Command/RoleCreatorCommand.php (hunk 2 above)
Comment (torqdev) [reply src/Command/RoleCreatorCommand.php:221]: Nice, I just tried it out and it looks like that should work.

Comment (lukemacausland) [reply src/Command/RoleCreatorCommand.php]: Nice I just tried that out, and it looks like that works as expected

**Pull request merged**